lang/java/avro/src/test/java/org/apache/avro/util/internal/TestJacksonUtils.java

```diff
@@ -102,4 +114,22 @@ void testToObject() {
     assertEquals("a", toObject(TextNode.valueOf("a"), SchemaBuilder.unionOf().stringType().and().intType().endUnion()));
   }
 
+  @ParameterizedTest
+  @MethodSource("nodes")
+  void cycle(JsonNode input) {
+    Object object = JacksonUtils.toObject(input);
+    JsonNode node = JacksonUtils.toJsonNode(object);
+    Assertions.assertEquals(input, node);
+  }
+
+  public static Stream<Arguments> nodes() {
+    ObjectNode o1 = JsonNodeFactory.instance.objectNode();
+    o1.put("intField", 123);
```

Comment:
How about small integers in a LongNode?

Note: parsing JSON text must actually create an IntNode in these cases, as otherwise parsing a logical type `decimal` will fail!

Reply:
Any suggestion for another test case here ?

Reply:
This is tricky. We can see in the tests that `int` and `long` are correctly preserved by the tests here, so I think/hope we can assume the test is complete.